src/status_im/transport/inbox.cljs

```diff
@@ -130,119 +92,138 @@
 
 (re-frame/reg-fx
  ::request-messages
  (fn [{:keys [wnode topics to from sym-key-id web3]}]
-   (request-messages web3
-                     wnode
-                     topics
-                     to
-                     from
-                     sym-key-id
-                     #(log/info "offline inbox: request-messages response" %)
-                     #(log/error "offline inbox: request-messages error" %1 %2 to from))))
+   (request-inbox-messages web3
+                           wnode
+                           topics
+                           to
+                           from
+                           sym-key-id
+                           #(log/info "offline inbox: request-messages response" %)
+                           #(log/error "offline inbox: request-messages error" %1 %2 to from))))
+
+(defn update-mailserver-status [transition {:keys [db]}]
+  (let [state transition]
+    {:db (assoc db :mailserver-status state)}))
+
+(defn generate-mailserver-symkey [{:keys [db] :as cofx}]
+  (when-not (:inbox/sym-key-id db)
+    {:shh/generate-sym-key-from-password {:password   (:inbox/password db)
+                                          :web3       (:web3 db)
+                                          :on-success (fn [_ sym-key-id]
+                                                        (re-frame/dispatch [:inbox/get-sym-key-success sym-key-id]))
+                                          :on-error   #(log/error "offline inbox: get-sym-key error" %)}}))
+
+(defn connect-to-mailserver
+  "Add mailserver as a peer using ::add-peer cofx and generate sym-key when it doesn't exists
+   Peer summary will change and we will receive a signal from status go when this is successful
+   A connection-check is made after `connection timeout` is reached and mailserver-status is
+   changed to error if it is not connected by then"
+  [{:keys [db] :as cofx}]
+  (let [web3     (:web3 db)
+        wnode    (get-current-wnode-address db)]
+    (when config/offline-inbox-enabled?
+      (handlers-macro/merge-fx cofx
+                               {::add-peer {:wnode wnode}
+                                :utils/dispatch-later [{:ms connection-timeout :dispatch [:inbox/connection-check]}]}
+                               (update-mailserver-status :connecting)
+                               (generate-mailserver-symkey)))))
+
+(defn peers-summary-change-fx
+  "Called when a peer summary signal is received or we want to try to connect to mailserver ie. after login.
+   Marks the mailserver as trusted peer if it is already added to the list of peers or connects to it if it's
+   not the case"
+  [{:keys [db] :as cofx}]
+  (if (and (:account/account db)
+           config/offline-inbox-enabled?)
+    (let [{:keys [peers-summary peers-count]} db]
+      (if (zero? peers-count)
+        (update-mailserver-status :disconnected cofx)
+        (let [wnode (get-current-wnode-address db)
```

Comment:
i would put all `let`s to a single form above condition forms (at least `ifs`), just to make it a bit easier to read